### ISSUE TITLE
feat: support atomic bulk actions

### ISSUE THREAD

**Issue by frankdugan3** [ash-project/ash_paper_trail#52]
**Is your feature request related to a problem? Please describe.**

I have a number of bulk actions that require auditing and atomic actions w/ rollback. For example, a packing list form which updates the ownership of a number of scanned items.

The trouble is that `AshPaperTrail` currently interferes with atomicity. `:full_diff` tracking mode may not be possible currently, but `:changes_only` should be possible to support atomically.

**Describe the solution you'd like**

The change for `create_new_version` should be refactored for support of bulk/atomic operations where possible. Currently, this would probably exclude `:full_diff` tracking mode, supporting only `:snapshot` and `:changes_only`.

**Describe alternatives you've considered**

- Mixins aren't capable of adding this as an escape hatch.
- Currently using a custom changeset to store changes in a []json column, which is pretty hacky.

**Comment by torazar** [comment]
I'm taking a look at this. Here's my plan. Refactor as desired to add bulk/atomic actions as a new setting, applicable to :snapshot and :changes. I'll keep :full_diff as non-atomic, and add a new execution strategy for atoms. How does this sound?

**Comment by zachdaniel** [comment]
It looks like maybe this has already been done 🤔 

```elixir
  @impl true
  def atomic(changeset, _opts, _context) do
    change_tracking_mode = AshPaperTrail.Resource.Info.change_tracking_mode(changeset.resource)

    if change_tracking_mode == :full_diff do
      {:not_atomic,
       "Cannot perform full_diff change tracking with AshPaperTrail atomically. " <>
         "You might want to choose a different tracking mode or set require_atomic? to false on your update actions."}
    else
      # Changes will be tracked in after_batch
      {:ok, changeset}
    end
  end
```

I had no idea how many issues we had open that were already resolved 😓

**Comment by frankdugan3** [comment]
Yeah, I can verify that by the time I finally got around to implementing that packing list, atomic is supported. I'm going to close it now assuming `:full_diff` remains intractable. Feel free to re-open if you want to make that part of the goal.
